### PR TITLE
Cu 2v8rzx9 blockchain integration   byo gas mamali

### PR DESCRIPTION
**Pull request by easteregg**
## Issue
https://app.clickup.com/t/2v8rzx9

## Description
BYO Gas implementation in Frontend

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2v8rzx9 Blockchain integration - BYO Gas](https://app.clickup.com/t/2v8rzx9)

**Review by dzmitry-lahoda — APPROVED** [review]

File: frontend/packages/substrate-react/src/dotsama/DotSamaContext.tsx [hunk 1]
```diff
@@ -95,6 +109,12 @@ export const DotSamaContextProvider = ({
     rpc: any;
     types: any;
   }[];
+  supportedRelaychains?: {
+    chainId: "kusama" | "polkadot";
```
Comment (saadjhk) [inline frontend/packages/substrate-react/src/dotsama/DotSamaContext.tsx:113]: can use `RelayChainId` here

File: frontend/packages/substrate-react/src/dotsama/hooks/useEagerConnect.ts [hunk 2]
```diff
@@ -36,7 +36,10 @@ export const useEagerConnect = (chainId: ParachainId): boolean => {
 
   useEffect(() => {
     if (selectedAccount) {
-      localStorage.setItem("selectedAccount", selectedAccount.address);
+      localStorage.setItem(
+        `selectedAccount-${chainId}`,
```
Comment (saadjhk) [inline frontend/packages/substrate-react/src/dotsama/hooks/useEagerConnect.ts:40]: @easteregg what's the intention of storing chain id as well?

**Review by saadjhk — APPROVED** [review]

File: frontend/packages/substrate-react/src/dotsama/DotSamaContext.tsx [hunk 3]
```diff
@@ -9,16 +9,29 @@ import {
...
-  InjectedExtension,
-} from "@polkadot/extension-inject/types";
+import type { InjectedAccount, InjectedAccountWithMeta, InjectedExtension, } from "@polkadot/extension-inject/types";
 import { decodeAddress, encodeAddress } from "@polkadot/util-crypto";
-import { createParachainApis } from "./utils";
+import { createParachainApis, createRelaychainApis } from "./utils";
 
 const truncate_regex = /^([a-zA-Z0-9]{4})[a-zA-Z0-9]+([a-zA-Z0-9]{4})$/;
 
+function getAccounts(
```
[the remark this answers is not in the document]
Comment (easteregg) [reply frontend/packages/substrate-react/src/dotsama/DotSamaContext.tsx:18]: No, this is a refactor of previously written code to re-use it for both Parachain and Relaychain initiation.

File: frontend/packages/substrate-react/src/dotsama/hooks/useEagerConnect.ts (hunk 2 above)
Comment (easteregg) [reply frontend/packages/substrate-react/src/dotsama/hooks/useEagerConnect.ts:40]: Eager connect failes to connect when we pass in other chains as well, mainly it fails to find the right wallet, so we need to have those values as well.

**Pull request merged**